src/openms/include/OpenMS/METADATA/ID/IdentificationData.h

```diff
@@ -0,0 +1,556 @@
+      const DataProcessingStep& step, SearchParamRef search_ref);
+
+    ScoreTypeRef registerScoreType(const ScoreType& score);
+
+    DataQueryRef registerDataQuery(const DataQuery& query);
+
+    ParentMoleculeRef registerParentMolecule(const ParentMolecule& parent);
+
+    void registerParentMoleculeGrouping(const ParentMoleculeGrouping& grouping);
+
+    IdentifiedPeptideRef registerIdentifiedPeptide(const IdentifiedPeptide&
+                                                   peptide);
+
+    IdentifiedCompoundRef registerIdentifiedCompound(const IdentifiedCompound&
+                                                     compound);
+
+    IdentifiedOligoRef registerIdentifiedOligo(const IdentifiedOligo& oligo);
+
+    QueryMatchRef registerMoleculeQueryMatch(const MoleculeQueryMatch& match);
+
+    MatchGroupRef registerQueryMatchGroup(const QueryMatchGroup& group);
+
+    const InputFiles& getInputFiles() const
+    {
+      return input_files_;
+    }
+
+    const DataProcessingSoftware& getDataProcessingSoftware() const
+    {
+      return processing_software_;
+    }
+
+    const DataProcessingSteps& getDataProcessingSteps() const
+    {
+      return processing_steps_;
+    }
+
+    const DBSearchParams& getDBSearchParams() const
+    {
+      return db_search_params_;
+    }
+
+    const DBSearchSteps& getDBSearchSteps() const
+    {
+      return db_search_steps_;
+    }
+
+    const ScoreTypes& getScoreTypes() const
+    {
+      return score_types_;
+    }
+
+    const DataQueries& getDataQueries() const
+    {
+      return data_queries_;
+    }
+
+    const ParentMolecules& getParentMolecules() const
+    {
+      return parent_molecules_;
+    }
+
+    const ParentMoleculeGroupings& getParentMoleculeGroupings() const
+    {
+      return parent_molecule_groupings_;
+    }
+
+    const IdentifiedPeptides& getIdentifiedPeptides() const
+    {
+      return identified_peptides_;
+    }
+
+    const IdentifiedCompounds& getIdentifiedCompounds() const
+    {
+      return identified_compounds_;
+    }
+
+    const IdentifiedOligos& getIdentifiedOligos() const
+    {
+      return identified_oligos_;
+    }
+
+    const MoleculeQueryMatches& getMoleculeQueryMatches() const
+    {
+      return query_matches_;
+    }
+
+    const QueryMatchGroups& getQueryMatchGroups() const
+    {
+      return query_match_groups_;
+    }
+
+    void addScore(QueryMatchRef match_ref, ScoreTypeRef score_ref,
+                  double value);
+
+
+    /*!
+      @brief Set a data processing step that will apply to all subsequent "register..." calls.
+
+      This step will be appended to the list of processing steps for all relevant elements that are registered subsequently (unless it is already the last entry in the list).
+      If a score type without a software reference is registered, the software reference of this processing step will be applied.
+
+      Effective until @ref clearCurrentProcessingStep() is called.
```

Comment:
Could this be done stateless? Current design harbors some potential for hard to debug code.